gateway-server/src/main/java/org/apache/knox/gateway/services/security/impl/ZookeeperRemoteAliasService.java

```diff
@@ -199,12 +205,17 @@ public void addAliasForCluster(final String clusterName,
             "Failed to store alias %s for cluster %s in remote registry", alias,
             clusterName));
       }
+    } else {
+      // Add it to the local cache only
+      Map<String, String> clusterAliases = aliasCache.computeIfAbsent(clusterName, m -> new ConcurrentHashMap<>());
```

Comment:
These 2 lines are used a few times. Might help to make it a "add to alias cache" method.